### ISSUE TITLE
ws_url seems undefined

### ISSUE THREAD

**Issue by tgranqvist** [huh-sters/traffcap#2]
Hello

ws_url seems undefined, as when I launch this app in Docker, I get the following error in the browser console:

```
index-BUki7xLF.js:21  WebSocket connection to 'wss://bin.my.tld/undefined/traffic/ws' failed: 
 Closed!
```

The container log says the following:
![image](https://github.com/user-attachments/assets/65953bcd-94dc-4d24-bc3b-50286ed15827)

I **think** I tracked the issue down to the VITE_WS_API_URL being empty. In `TrafficView` we launch a web socket connection to `ws_url`, which is imported from `@/repositories/repository`. There it is set as follows: `export const ws_url: string | undefined = import.meta.env.VITE_WS_API_URL`. What should this value be? Should I set it in my Composefile?

**Comment by Centurix** [comment]
You're correct, there are three things that need defining for the UI to work. They are generally set in envrionment variables. The easiest way is to create a `.env` file the the same directory as the `package.json` file and put the following into it:

```
VITE_WS_API_URL=http://0.0.0.0:9669
VITE_API_URL=http://0.0.0.0:9669
BASE_URL=http://0.0.0.0:9669
```

If you're building using `npm` then these will be picked up and used in the frontend.

Thanks for trying this out, I haven't worked on this in a while and the project was stalled while SQLModel was sorting out a couple of issues around primary key settings, Pydantic, PyDANJA and SQLAlchemy. I just had a quick look and these issues appear to be resolved.

**Comment by tgranqvist** [comment]
Thanks for swift reply. Great to hear upstream issues have been resolved! I am interested in contribution to this project, if you are open to collaborators.

I'm not building with npm but running from the Docker image provided. Those variables are not listed there, so didn't include them in my composefile. Setting them to the values you gave above removed one error, but it's still not a 100 % working. The websocket now connects (yay 👍 ), but then I see a 404 on the GET to `https://bin.my_server.tld/undefined/traffic`. It's on line 21 in the minified code: ` l(ce.url, {...`.

I'm running this on my home server fronted by a Cloudflare tunnel connection. Do I need to account for this somehow with these variables?

**Comment by Centurix** [comment]
No problems, I'm open to collaborators on this project for sure. There's a very rough roadmap in the readme which is my immediate list of things to get working. If there's momentum, then I might move to something in Trello or Planka.

As for this issue, I'll take a look as I may have just flat out broken it with my last modifications!

**Comment by Centurix** [comment]
Managed to move on this and get the docker image running correctly. It should work out of the box now.
